django/forms/widgets.py

```diff
@@ -1070,18 +1070,16 @@ def value_from_datadict(self, data, files, name):
         if y == m == d == '':
             return None
         if y is not None and m is not None and d is not None:
-            if settings.USE_L10N:
-                input_format = get_format('DATE_INPUT_FORMATS')[0]
-                try:
-                    date_value = datetime.date(int(y), int(m), int(d))
-                except ValueError:
-                    pass
-                else:
-                    date_value = datetime_safe.new_date(date_value)
-                    return date_value.strftime(input_format)
-            # Return pseudo-ISO dates with zeros for any unselected values,
-            # e.g. '2017-0-23'.
-            return '%s-%s-%s' % (y or 0, m or 0, d or 0)
+            input_format = get_format('DATE_INPUT_FORMATS')[0]
+            try:
+                date_value = datetime.date(int(y), int(m), int(d))
+            except ValueError:
+                # Return pseudo-ISO dates with zeros for any unselected values,
+                # e.g. '2017-0-23'.
+                return '%s-%s-%s' % (y or 0, m or 0, d or 0)
+            else:
+                date_value = datetime_safe.new_date(date_value)
+                return date_value.strftime(input_format)
```

Comment:
Minor but the `else` clause wrapping is unnecessary here given you return in the `except`.

Reply:
It is necessary. When try doesn't throw an exception, the else block is executed.

Reply:
@CruxBox my point is that you are already returning in the case of an exception *and* in the `else` clause so the code is equivalent to

```python
try:
    date_value = datetime.date(int(y), int(m), int(d))
except ValueError:
    return '%s-%s-%s' % (y or 0, m or 0, d or 0)
date_value = datetime_safe.new_date(date_value)
return date_value.strftime(input_format)
```

Again, this is minor but we usually avoid using `else` in the rest of the code base unless it's necessary.

Reply:
Oh right. Sorry I misinterpreted what you said earlier! Want me to remove the `else` then?